back/app/services/permissions_service.rb

```diff
@@ -144,7 +144,7 @@ def requirements_mapping
         users[:built_in][:email] = 'require'
         required_field_keys = CustomField.registration.required.map(&:key)
         users[:custom_fields].each_key do |key|
-          users[:custom_fields][key] = required_field_keys.include? key ? 'require' : 'ask'
+          users[:custom_fields][key] = (required_field_keys.include?(key) ? 'require' : 'ask')
```

Comment:
Any reason for introducing brackets here?

Reply:
This code didn't work before. Very likely the assignment had precedence over the if statement, and the same value would be set for all custom fields, regardless of whether they were required or not. Adding parentheses here fixed it.